numba/cuda/functions.cu

```diff
@@ -0,0 +1,211 @@
+#include "cuda_fp16.h"
+
+extern "C" __device__ int
+hdiv_wrapper(
+  short* return_value,
+  short x,
+  short y
+)
+{
+  __half retval = __hdiv(__short_as_half (x), __short_as_half (y));
+  
+  *return_value = __half_as_short (retval);
+  // Signal that no Python exception occurred
+  return 0;
+}
+
+extern "C" __device__ int
+hsin_wrapper(
```

Comment:
Although C preprocessor macros are generally to be avoided where possible, I can't help but think it'd be a worthwhile tradeoff to implement a macro for all the unary function wrappers, rather than duplicating lots of boilerplate. What do you think?

Alternatively perhaps a template function with multiple explicit instantiations would be nicer, but I'm not sufficiently experience in using NVRTC and/or extern C functions to see ahead of trying it whether that could actually work in this situation - could that be worth a try?

Reply:
Done.